Additonal wxWebView fixes and additions

TcT2k opened the pull request:
* Add fullscreen event (when javascript triggers fullscreen)
* Add find for edge and WKWebView backend
* Fix [#19074](https://trac.wxwidgets.org/ticket/19074)
* Don't trigger erroneous wxEVT_WEBVIEW_ERROR when vetoing wxEVT_WEBVIEW_NAVIGATING  with edge

TcT2k committed:
Handle in-document navigation with wxWebViewEdge

Send EVT_WEBVIEW_NAVIGATING, EVT_WEBVIEW_NAVIGATED and EVT_WEBVIEW_LOADED
when an anchor is clicked within a document. This kind of navigation
cannot be vetoed.

Fixes: #19074

TcT2k committed:
DPI independent default size in WebView sample

TcT2k committed:
Add wxEVT_WEBVIEW_FULLSCREEN_CHANGED event

The event is triggered by wxWebView when the page requests fullscreen view.

It's currently only implemented for the edge backend.

TcT2k committed:
Don't send wxWebViewEdge cancel event on veto

When vetoing a wxEVT_WEBVIEW_NAVIGATING a wxEVT_WEBVIEW_ERROR was send
this was unexpected and unnecessary.

TcT2k committed:
Implement wxEVT_WEBVIEW_FULLSCREEN_CHANGED in GTK

vadz commented on interface/wx/webview.h:
```diff
@@ -428,6 +428,10 @@ class wxWebViewHandler
+    @event{EVT_WEBVIEW_FULL_SCREEN_CHANGED(id, func)}
+       Process a @c EVT_WEBVIEW_FULL_SCREEN_CHANGED event, generated when
+       the page wants to enter or leave fullscreen. Use GetInt to get the status.
+       Currently only implemented for the edge and WebKit2GTK+ backend.
```
```suggestion
       Currently only implemented for the edge and WebKit2GTK+ backend
       and is only available in wxWidgets 3.1.5 or later.
```

vadz commented on src/msw/webview_edge.cpp:
```diff
@@ -278,6 +302,22 @@ HRESULT wxWebViewEdgeImpl::OnContentLoading(ICoreWebView2* WXUNUSED(sender), ICo
+        return hr;
+
+    wxWebViewEvent event(wxEVT_WEBVIEW_FULLSCREEN_CHANGED, m_ctrl->GetId(),
+        m_ctrl->GetCurrentURL(), "");
```
Nitpick
```suggestion
        m_ctrl->GetCurrentURL(), wxString());
```

vadz reviewed (commented):
Thanks!

I'll merge this when the CI builds finish (unless you prefer to keep it open if you plan to add more commits to this branch).

TcT2k committed:
Apply suggestions from code review

Co-authored-by: VZ <vz-github@zeitlins.org>

TcT2k commented:
> Thanks!
> 
> I'll merge this when the CI builds finish (unless you prefer to keep it open if you plan to add more commits to this branch).

Don't plan more changes here, should be good to merge, when CI finishes. I applied your suggestions.

MaartenBent commented:
Based on the diff it looks like you changed all line endings to crlf. I should extend the EOL check with checking for expected EOL.

vadz commented:
> Based on the diff it looks like you changed all line endings to crlf. I should extend the EOL check with checking for expected EOL.

Yes, I'd like to suggest to use the list of extensions I've posted in a comment before (maybe removing some that we don't use) and checking that EOLs correspond to the expected ones and not just that they're consistent.

TcT2k committed:
Provide generic wxWebView::Find() implementation

This implementation is based on javascript 'window.find()'
https://developer.mozilla.org/en-US/docs/Web/API/Window/find
It's not standard, but implemented in most browsers.

It's not a complete implementation as it can't determine how
many search results can be found.

If the backends API provides a native find interface it should be
used instead of this implementation, but it's good fallback.

vadz commented:
Thanks again, merging now!

P.S. The [memory leak here](https://github.com/wxWidgets/wxWidgets/pull/2227/checks?check_run_id=1875403286#step:9:59) seems to be real, i.e. it has happened more than once, although I have no idea why does it happen only sometimes (i.e. I wasn't able to reproduce it). This is unrelated to the changes here, of course.
